src/db/sysdb_search.c

```diff
@@ -218,6 +218,21 @@ int sysdb_getpwnam(TALLOC_CTX *mem_ctx,
         goto done;
     }
 
+    if (res->count > 1) {
+        /* We expected either 0 or 1 result for search with
+         * SYSDB_PWNAM_FILTER, but we got more. This error
+         * is handled individually depending on what function
+         * called sysdb_getpwnam, so we just print a message
+         * here and let the caller decide what error code to
+         * propagate based on res->count > 1. */
+        DEBUG(SSSDBG_CRIT_FAILURE,
+              "Search for [%s] returned multiple results. It can be an email "
+              "address shared among multiple users or an email address of a "
+              "user that conflicts with another user's fully qualified name. "
+              "SSSD will not be able to handle those users properly.\n",
+              sanitized_name);
+    }
+
```

Comment:
This function use following filter: `SYSDB_PWNAM_FILTER`
Which is expanded into `"(&("SYSDB_UC")(|("SYSDB_NAME_ALIAS"=%s)("SYSDB_NAME_ALIAS"=%s)("SYSDB_NAME"=%s)))"`

Could you explain how we can be related to email address which is mentioned in critical debug message?
Because title of ticket https://fedorahosted.org/sssd/ticket/3293 is
"SSSD authentication fails when two IPA accounts share an email address"

Reply:
I explained it in my previous comment.

Reply:
Sorry but it was not properly explained.

Because `SYSDB_USER_EMAIL` ("mail") is not the same as `SYSDB_NAME_ALIAS` ("nameAlias")

Reply:
I am confused by your last comment. Do you need further explanation or not?

Reply:
Yes, I need because problem is with `SYSDB_USER_EMAIL` as you can see in rhbz
https://bugzilla.redhat.com/show_bug.cgi?id=1408294#c7
https://bugzilla.redhat.com/show_bug.cgi?id=1408294#c8

And I cannot see a way how we can get two user the same EMAIL in `sysdb_getpwnam`

Reply:
As I said in the previous comments in this PR. The problem is that users have the same email in LDAP (IPA in the BZ). We fetch the attribute from LDAP and store it TWICE in the local SYSDB database. Once as the SYSDB_USER_EMAIL (in the original case) and once lowercased in the SYSDB_NAME_ALIAS, because we want the search by email address to be case insensitive.

The user in the BZ hits it during the getpwnam search, because several users have the same SYSDB_NAME_ALIAS (generated by lowercasing the EMAIL)

Reply:
And that is a wrong misuse of `nameAlias` IMHO. Because then we need to guess why more entries were returned.
It would me much better to store it as `emailAlias`.

The problem with current message is that is really vague and will not help user to find problematic entries.
If we have more entries with the same GID then it is quite trivial to find collision in sysdb cache. Because attribute for entry is unique enough.

Let say that we have "user123" with email "test@example.com" and "benutzer123" with email "TEST@example.com". Your debug message will not help to find problematic users.

Reply:
I do not think it is a misuse of nameAlias, but having emailAlias could have some benefits. That discussion is however out of scope for this PR and refactoring of how we handle the emails in sysdb would also require update of sysdb version.

You say that the message is vague and will not help the user. I do not see it that way, but please suggest better wording. Saying that it is vague is not giving me any hints on how to improve it.

In your example with user123" with email "test@example.com" and "benutzer123" with email "TEST@example.com", it is clear conflict of emails. I do not see how is the message not helpful here. It clearly says that there may be users with conflicting emails.